libraries/Microsoft.Bot.Builder.Dialogs.Declarative/Converters/InterfaceConverter.cs

```diff
@@ -101,13 +114,28 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
 
                     var tokenToBuild = TryAssignId(jToken, sourceContext);
 
-                    T result = this.resourceExplorer.BuildType<T>(kind, tokenToBuild, serializer);
+                    T result;
+                    if (refDialogName != null && cachedRefDialogs.ContainsKey(refDialogName))
+                    {
+                        result = cachedRefDialogs[refDialogName];
+                    }
+                    else
+                    {
+                        result = this.resourceExplorer.BuildType<T>(kind, tokenToBuild, serializer);
+                        if (refDialogName != null)
+                        {
+                            cachedRefDialogs.Add(refDialogName, result);
+                        }
+                    }
 
                     // Associate the most specific source context information with this item
                     if (sourceContext.CallStack.Count > 0)
                     {
                         range = sourceContext.CallStack.Peek().DeepClone();
-                        DebugSupport.SourceMap.Add(result, range);
+                        if ((DebugSupport.SourceMap as Dictionary<object, SourceRange>).ContainsKey(result))
```

Comment:
Why not use the "TryGetValue" in SourceMap,
```
if(!DebugSupport.SourceMap.TryGetValue(result, out var _))
{
   DebugSupport.SourceMap.Add(result, range);
}
```

Reply:
good suggestion. fixed.